tests/Authentication/AuthenticationIntegrationTest.php

```diff
@@ -241,4 +242,13 @@ public function testFetchTokensWithNullCredentialsThrowsException()
         $authenticator = new Authenticator($this->guzzle);
         $authenticator->getAuthHeader();
     }
+
+    public function testAuthenticationExceptionThrown()
```

Comment:
I don't _love_ this test name and am open to suggestion :)

Reply:
`testAuthorizationExceptionWhenUnauthorizedStatusCode`?  I tend to lean towards longer test names to try to describe what happens and when.